billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/documents/PTGenericInvoiceIssuingHandler.java

```diff
@@ -110,13 +112,17 @@ protected <D extends AbstractDAOPTGenericInvoice<T>> T issue(final T document, f
         String sourceHash = GenerateHash.generateSourceHash(invoiceDate, systemDate, formattedNumber,
                 document.getAmountWithTax(), previousHash);
 
+		if (invoiceSeriesEntity.getSeriesUniqueCode().isPresent()) {
+			validateSeriesUniqueCode(invoiceSeriesEntity.getSeriesUniqueCode().get());
+		}
+
         final String atcud = invoiceSeriesEntity
             .getSeriesUniqueCode()
             .map(s -> new StringBuilder()
                 .append(s)
                 .append("-")
-                .append(seriesNumber.toString()))
-            .orElse(new StringBuilder().append("0"))
+                .append(seriesNumber))
+            .orElseThrow(() -> new SeriesUniqueCodeNotFilled(invoiceSeriesEntity.getSeries()))
```

Comment:
At the moment we are currently using the series as the only content of the message on SeriesUniqueCodeNotFilled instances, can we provide a more complete contextual message?